py/dml/types.py

```diff
@@ -784,16 +867,26 @@ def sizeof(self):
         if elt_size == None:
             return None
         return self.size.value * elt_size
+
     def cmp(self, other):
-        if compat.dml12_misc in dml.globals.enabled_compat:
-            if isinstance(other, (TArray, TPtr)):
-                return self.base.cmp(other.base)
-        elif isinstance(other, (TPtr, TArray)):
-            if self.base.void or other.base.void:
-                return 0
-            if self.base.cmp(other.base) == 0:
-                return 0
-        return NotImplemented
```

Comment:
This is a strange one. Previously, `TArray.cmp` 's logic caused it to be more lenient _without_ the `dml12_misc` compatibility feature. I originally thought that maybe you made a mistake while making #209, but no, that was true even before then. Same story with `TPtr.cmp`.
While moving old behaviors of `cmp` to `cmp_fuzzy` I've just dropped these `dml12_misc` check of `TArray` and `TPtr`.